src/cljs/vanilla/modal.cljs

```diff
@@ -0,0 +1,113 @@
+(ns vanilla.modal
+  (:require
+    [re-frame.core :as rf]
+    [day8.re-frame.tracing :refer-macros [fn-traced]]
+    [reagent.core :as r]))
+
+
+
+
+;;;;;;;;;;;;;;;;;;;;
+;
+;  Helper defs -
+;      The declared defs below are used to replace hiccup to make future changes easier if
+;      we need to change anything about the hiccup. Also they prevent typos throughout this namespace
+;
+;
+
+(def modal-start
+  :div.modal)
+
+(def modal-background
+  :div.modal-background)
+
+(def modal-card
+  :div.modal-card)
+
+;;;;;;;;;;;
+
+
+;;;;;;;;;;;
+;
+;   Modal Components
+;        The functions below return the sections of the modal as dictated in their function name.
+;        The idea is to break the modal into a header body and footer and handle each of them separately.
+;
+
+(defn modal-header
+  [title is-active]
+  [:header.modal-card-head
+   [:p.modal-card-title title]
+   [:button.delete {:aria-label "close"
+                    :on-click   #(reset! is-active false)}]])
+
+;; This is mostly a helper def
+(def modal-body-section
+  :section.modal-card-body)
+
+(defn modal-footer
+  ""
+  [enabled-button button-fn button-text is-active]
+  [:footer.modal-card-foot
+   [:button.button.is-success
+    {:disabled (not enabled-button)
+     :on-click #(do
+                  (button-fn)
+                  (reset! is-active false))}
+    button-text]
+   [:button.button {:on-click #(reset! is-active false)} "Cancel"]])
+
+
+;;;;;;;
+
+
+(defn create-modal
```

Comment:
since it's intended for use inside hiccup, which is "composed of reagent components," let's just make this called "modal," i.e., (defn modal ...). This also means the very first line of the comment (69) would be "This function returns a 'modal' Reagent component, using the Bulma.io "modal" css…"

Reply:
I named this "create-modal" so it didn't have the same name as it's namespace. Should we name functions the same as their namespaces? It seems like most of the clojure stuff I've read avoids this, although they don't condemn it either.  

We can change the namespace to modal-utilities or something.

Either way I will change the comment on line 69 to reflect that this is a Reagent component.

Reply:
Have a look at how re-com does it: https://github.com/day8/re-com/tree/master/src/re_com.

Reply:
A way to “get around” the modal/modal constrict is to use “:refer [modal]“ in the dependency rather than the :as clause, that way you just have the function as of it were in your current namespace.

Reply:
I think you're thinking of aliases, which makes sense to make it not used as modal/modal.

I was just talking about naming conventions within our project. I figured we would not want overlap but I'm picking up the hint that it doesn't matter haha. I'll change the name of the modal function to just "modal" and use :refer for alias work in other classes.